src/utils/Meta.js

```diff
@@ -1,60 +1,88 @@
 import _ from 'lodash'
+import * as stardust from 'stardust'
+
+const LIBRARIES = {
+  semanticUI: 'Semantic UI',
+  stardust: 'Stardust',
+}
+
+const TYPES = {
+  addon: 'addon',
+  global: 'global',
+  collection: 'collection',
+  element: 'element',
+  view: 'view',
+  module: 'module',
+}
+
+const LIBRARY_VALUES = _.values(LIBRARIES)
+const TYPE_VALUES = _.values(TYPES)
+const NAME_VALUES = _.keys(stardust)
+
+/**
+ * Determine if an object qualifies as a META object.
+ * It must have the required keys and valid values.
+ * @private
+ * @param {Object} _meta A proposed Stardust _meta object.
+ * @returns {Boolean}
+ */
+const isMeta = (_meta) => (
+  _.includes(LIBRARY_VALUES, _.get(_meta, 'library')) &&
+  _.includes(TYPE_VALUES, _.get(_meta, 'type')) &&
+  _.includes(NAME_VALUES, _.get(_meta, 'name'))
+)
+
+/**
+ * Extract the Stardust _meta object and optional key.
+ * Handles literal _meta objects, classes with _meta, objects with _meta
+ * @private
+ * @param {string|function|object} arg A class, a component instance, or meta object..
+ * @param {string} [key] A key to pluck from the _meta object.
+ * @returns {object|string|undefined}
+ */
+const getMeta = (arg, key) => {
+  let _meta = {}
+
+  // literal
+  if (isMeta(arg)) _meta = arg
+
+  // from prop
+  else if (isMeta(_.get(arg, '_meta'))) _meta = arg._meta
+
+  // from class
+  else if (isMeta(_.get(arg, 'constructor._meta'))) _meta = arg.constructor._meta
+
+  // default values
+  return key ? _meta[key] : _meta
+}
```

Comment:
You can pass essentially anything to this method and get a valid _meta object back if it could be resolved.  Otherwise, an empty object.